File: codewars/python/shortest_word.py
# https://www.codewars.com/kata/57cebe1dc6fdc20c57000ac9/train/python


def find_short(s):
    return min([len(i) for i in s.split()])


if __name__ == "__main__":
    s = input()
    print(find_short(s))

File: codewars/rust/old/shortest_word.rs
// https://www.codewars.com/kata/57cebe1dc6fdc20c57000ac9/train/rust


use std::io;

fn find_short(s: &str) -> u32 {
    let words: Vec<&str> = s.split(" ").collect();
    let mut count: Vec<usize> = Vec::new();

    for i in words.iter() {
        count.push(i.len());
    }

    let min = count.iter().min().unwrap();
    *min as u32
}

fn main() {
    let mut s = String::new();

    io::stdin()
        .read_line(&mut s)
        .expect("Failed to read line");

    println!("{}", find_short(&s));
}
